### PR TITLE
feat(amm): add swap extrinsic

### LINKED ISSUE #533: AMM: Implement swap extrinsic
Implement AMM module `swap` method for making trades.

### PR DESCRIPTION
**Pull request by peterslany**
Resolves #533

### CODE REVIEW

File: src/parachain/amm.ts [hunk 1]
```diff
@@ -420,14 +423,62 @@ export class DefaultAMMAPI implements AMMAPI {
...
+        );
+
+        await this.transactionAPI.sendLogged(swapExtrinsic, this.api.events.zenlinkStableAmm.CurrencyExchange, true);
+    }
+
     async swap(
         trade: Trade,
         minimumAmountOut: MonetaryAmount<CurrencyExt>,
-        recipient: AddressOrPair,
+        recipient: string,
```
Comment (bvotteler) [inline src/parachain/amm.ts:473]: Does the recipient string represent an Account?
If so, I would prefer if it was typed more strongly.

File: src/parachain/amm/encoding.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    )[];
+} {
+    const amountIn = trade.inputAmount._rawAmount.toString();
+    const amountOutMin = minimumAmountOut._rawAmount.toString();
+    const path = trade.path.map((pathElement) =>
+        isStableMultiPathElement(pathElement)
+            ? {
+                  Stable: {
+                      poolId: pathElement.pool.poolId,
+                      basePoolId: isStableMetaMultiPathElement(pathElement) && pathElement.basePool.poolId,
```
Comment (bvotteler) [inline src/parachain/amm/encoding.ts:58]: What happens if `isStableMetaMultiPathElement()` returns false? 
Is any falsy value valid for `basePoolId`?

File: src/parachain/amm.ts (hunk 1 above)
Comment (peterslany) [reply src/parachain/amm.ts]: Yes, it represent an account, updated :+1:

File: src/parachain/amm/encoding.ts (hunk 2 above)
Comment (peterslany) [reply src/parachain/amm/encoding.ts]: I think this was mistake, good catch! Zenlink sdk returns undefined here, but now I changed it to explicit 0.

**Review by bvotteler — APPROVED** [review]
LGTM

**Pull request merged**